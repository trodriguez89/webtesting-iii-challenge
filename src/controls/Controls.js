import React from 'react';

const Controls = props => {
  const { locked, closed, toggleLocked, toggleClosed } = props;

  return (
    <div className="controls panel">
      <button data-testid="locked" disabled={!closed} onClick={toggleLocked} className="toggle-btn">
        {locked ? 'Unlock Gate' : 'Lock Gate'}
      </button>
      <button data-testid="closed" disabled={locked} onClick={toggleClosed} className="toggle-btn">
        {closed ? 'Open Gate' : 'Close Gate'}
      </button>
    </div>
  );
};

export default Controls;
