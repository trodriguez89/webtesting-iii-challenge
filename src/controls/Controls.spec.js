import React from "react";
import { render } from "@testing-library/react";

import Controls from "./Controls";

// Test away!

test("Controls matches snapshot", () => {
    const tree = render(<Controls />)
    expect(tree.baseElement).toMatchSnapshot();
})

test("toggle closed", () => {
    const { getByTestId }= render(<Controls />)
    getByTestId("locked")
})

test("toggle closed", () => {
    const { getByTestId } = render(<Controls />)
    getByTestId("closed")
})