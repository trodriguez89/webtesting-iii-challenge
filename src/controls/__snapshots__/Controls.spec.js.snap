// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls matches snapshot 1`] = `
<body>
  <div>
    <div
      class="controls panel"
    >
      <button
        class="toggle-btn"
        data-testid="locked"
        disabled=""
      >
        Lock Gate
      </button>
      <button
        class="toggle-btn"
        data-testid="closed"
      >
        Close Gate
      </button>
    </div>
  </div>
</body>
`;
